src/datasources/cache/redis.cache.service.spec.ts

```diff
@@ -165,6 +182,25 @@ describe('RedisCacheService', () => {
     expect(redisClientTypeMock.quit).toHaveBeenCalledTimes(0);
   });
 
+  it('Deleting key calls delete and sets invalidationTime', async () => {
+    jest.useFakeTimers();
+    const now = jest.now();
+    const key = faker.string.alphanumeric();
+
+    await redisCacheService.deleteByKey(key, true);
+
+    expect(redisClientTypeMock.unlink).toHaveBeenCalledTimes(1);
+    expect(redisClientTypeMock.hGet).toHaveBeenCalledTimes(0);
+    expect(redisClientTypeMock.hSet).toHaveBeenCalledTimes(1);
+    expect(redisClientTypeMock.hSet).toHaveBeenCalledWith(
+      `invalidationTimeMs:${key}`,
+      '',
+      now.toString(),
+    );
+    expect(redisClientTypeMock.hDel).toHaveBeenCalledTimes(0);
+    expect(redisClientTypeMock.quit).toHaveBeenCalledTimes(0);
+  });
```

Comment:
We can also add an assertion which verifies that the key was set with a given TTL.

Reply:
Checks added in [b92584c](https://github.com/safe-global/safe-client-gateway/pull/970/commits/b92584cf5c50470b34305c8c9afffcc884c9fb21)